src/lib.rs

```diff
@@ -941,6 +943,36 @@ pub mod invariant {
         fn get_fee_tiers(&self) -> Vec<FeeTier> {
             self.fee_tiers.get_all()
         }
+
+        #[ink(message)]
+        fn get_tickmap(
+            &self,
+            pool_key: PoolKey,
+            starting_chunk: u16,
+            finishing_chunk: u16,
+        ) -> Vec<(u16, u64)> {
```

Comment:
There is no need for pagination here. We need the tickmap only to simulate swaps. In the interface, you should pass center tick index. It should return the maximum number of ticks symmetrically in the current chunk; half below and half above. If there are not enough ticks on one side, get more from the opposite side